pkg/apis/provisioning/v1alpha5/requirements.go

```diff
@@ -202,27 +202,30 @@ func (r Requirements) CapacityTypes() stringsets.String {
 }
 
 // Validate validates the feasibility of the requirements.
-func (r Requirements) Validate() (errs *apis.FieldError) {
-	for i, requirement := range r.Requirements {
+//gocyclo:ignore
+func (r Requirements) Validate() (errs []string) {
```

Comment:
I feel pretty strongly that we should model these as true errors. We can model this plainly using multierr as well.
```suggestion
func (r Requirements) Validate() (error) {
```

Reply:
What's your opinion about keeping it consistent with
https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go

I am open to either way.

Reply:
apimachinery's requirements are different, as they need to be careful about their dependency closure. They expect many packages to rely on them, so adding something like multierr results in huge dependency exposure. This problem doesn't exist for us, and all other validators in the codebase return `error`.